### PR TITLE
Feature/authorization

### PR DESCRIPTION
**Pull request by Jeffrey-Zutt**
First version of authorization.

- Login
- Logout
- Session timeout

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -2,3 +2,9 @@
 # https://create-react-app.dev/docs/adding-custom-environment-variables/
 
 REACT_APP_GATEWAY=http://localhost:8080/api/v1/
+
+# To login using grip:
+# REACT_APP_AUTH_URL="https://auth.grip-on-it.com/v2/rjsfm52t/oidc/idp/authorize?response_type=code&scope=openid&client_id=d3d664c7-bb33-4bf0-b7c9-b8bdf1199b76&redirect_uri=http://localhost:2999/"
+REACT_APP_AUTH_URL=/login/callback/?code=1234567890abcdefghijkl
```
Comment (gilleswittenberg) [inline .env.development:8]: Think `?code=1234567890abcdefghijkl` is added by Grip

Comment (Jeffrey-Zutt) [reply .env.development:8]: You can use this URL (a direct link to the callback), to bypass GRIP.

File: src/app/features/login/components/pages/callback/CallbackPage.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  useEffect(() => {
+    axios
+      .post(OIDCAuthUrl, { code })
+      .then(response => {
+        setToken(response.data.access)
+        return navigate(to("/"))
+      })
+      .catch(error => console.log("Something went wrong", error))
+  }, [ code ])
```
Comment (gilleswittenberg) [inline src/app/features/login/components/pages/callback/CallbackPage.tsx:27]: Cancel on unmount

File: src/app/features/login/components/pages/logout/LogoutPage.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      //     url: process.env.REACT_APP_AUTH_LOGOUT_URL,
+      //     headers: { mode: "no-cors", credentials: "include" }
+      //   })
+      // } finally {
+        await navigate(to("/login"))
+      // }
+    })()
+  }, [])
+
+  return <></>
```
Comment (gilleswittenberg) [inline src/app/features/login/components/pages/logout/LogoutPage.tsx:25]: `return null`?

File: src/app/features/shared/routing/Router.tsx [hunk 4]
```diff
@@ -2,12 +2,24 @@ import React from "react"
...
   <ReachRouter>
     {
+      // Pages that do NOT match the allowList are protected
+      Object
+        .entries(routes)
+        .filter(([path]) => !path.match(allowList))
+        .map(([path, Page]) => <ProtectedRoute page={Page} key={path} path={path} />)
+    }
+    {
+      // Pages that do match the whiteList are NOT protected
```
Comment (gilleswittenberg) [inline src/app/features/shared/routing/Router.tsx:19]: `whitelist` => `allowList` :)

**Pull request merged**